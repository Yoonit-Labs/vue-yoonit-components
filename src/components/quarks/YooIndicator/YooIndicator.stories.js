import yooIndicator from './YooIndicator.vue'
import PropsConfig from './YooIndicator.config'

export default {
  title: 'Yoonit/Indicator',
  component: yooIndicator,
  argTypes: {
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    },
    size: {
      control: {
        type: 'select',
        options: PropsConfig.size.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooIndicator },
  template:
    '<yoo-indicator v-bind="$props"></yoo-indicator>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Indicator test component' }

export const Default = Template.bind({})
Default.args = { text: '1' }

export const Primary = Template.bind({})
Primary.args = { text: '1', fill: 'primary' }

export const Light = Template.bind({})
Light.args = { text: '2', fill: 'light' }

export const Dark = Template.bind({})
Dark.args = { text: '10', fill: 'dark' }

export const Neutral = Template.bind({})
Neutral.args = { text: '5', fill: 'neutral' }
