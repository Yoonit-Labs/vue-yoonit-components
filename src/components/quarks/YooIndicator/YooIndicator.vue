<template lang="pug">
  div(
    :class="['yoo-indicator', ...takeModifier]"
  ) {{ text }}
</template>

<script>

export default {
  name: 'YooIndicator',
  props: {
    fill: {
      type: String,
      required: false,
      default: 'danger',
      validator: value =>
        [
          'primary',
          'danger',
          'light',
          'dark',
          'neutral'
        ]
          .includes(value)
    },
    size: {
      type: String,
      required: false,
      default: 'normal',
      validator: value =>
        [
          'small',
          'medium',
          'normal',
          'large'
        ]
          .includes(value)
    },
    animation: {
      type: Boolean,
      required: false,
      default: true
    },
    text: {
      type: String,
      required: false,
      default: ''
    }
  },
  components: {},
  data: () => ({}),
  beforeCreate () {},
  created () {},
  beforeMount () {},
  mounted () {},
  beforeUpdate () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  computed: {
    takeModifier () {
      const classList = [`yoo-indicator--${this.size}`, `yoo-indicator--${this.fill}`]

      if (this.animation) {
        classList
          .push('yoo-indicator--pulse')
      }

      return classList
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooIndicator.sass" lang="sass" scoped></style>
