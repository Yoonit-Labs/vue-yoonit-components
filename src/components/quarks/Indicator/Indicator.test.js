import { shallowMount } from '@vue/test-utils'
import YooIndicator from '@/components/quarks/Indicator/Indicator.vue'
import PropsConfig from '@/components/quarks/Indicator/Indicator.config'

const classBlock = 'yoo-indicator'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooIndicator, {
    slots: { default: SlotText }
  })
}

describe('YooIndicator Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.find('.yoo-indicator').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('fill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fill.options.includes(YooIndicator.props.fill.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(fill => {
        it(`Includes fill class: .${classBlock}--${fill}`, async () => {
          await wrapper.setProps({ fill })
          expect(wrapper.find(`.${classBlock}--${fill}`).exists()).toBe(true)
        })
      })
    })
    describe('size', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.size.options.includes(YooIndicator.props.size.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(size => {
        it(`Includes size class: .${classBlock}--${size}`, async () => {
          await wrapper.setProps({ size })
          expect(wrapper.find(`.${classBlock}--${size}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
})
