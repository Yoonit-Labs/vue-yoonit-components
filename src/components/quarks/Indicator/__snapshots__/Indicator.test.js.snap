// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooIndicator Component Matches Snapshot 1`] = `<div class="yoo-indicator yoo-indicator--normal yoo-indicator--danger yoo-indicator--pulse"></div>`;
